HOLMS.Messaging/MessagingConfiguration.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+
+namespace HOLMS.Messaging {
+    public class MessagingConfiguration {
+        public string Host { get; protected set; }
+        public ushort Port { get; protected set; }
+        public string User { get; protected set; }
+        public string Password { get; protected set; }
+        public string VHost { get; protected set; }
+
+        public MessagingConfiguration(string connectionString) {
+            //example: amqp://user:pass@hostname:port/vhost
+            var protocolSplit = connectionString.Split(new string[] { "://" }, 2, StringSplitOptions.None);
+            if (protocolSplit.Length != 2) {
+                throw new ArgumentException($"No protocol specified in string {connectionString}");
+            }
+            var protocol = protocolSplit[0];
+            if (protocol != "amqp" && protocol != "amqps") {
+                throw new ArgumentException($"Invalid protocol {protocol}");
+            }
+            var connection = protocolSplit[1];
+            if (connection.Contains("@")) {
+                var allCredentials = connection.Split('@')[0].Split(new string[] { ":" }, 2, StringSplitOptions.None);
+                //Allow colons in password, just not in username
+                User = allCredentials[0];
+                if (allCredentials.Length > 1) {
+                    Password = allCredentials[1];
+                }
+                //strip the credentials from the string. Assumes only one "@" in the entire string, and it can't be in the password
```

Comment:
We could adjust the code to get the last "@" character. There shouldn't be one in the hostname, port, or vhost